src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java

```diff
@@ -0,0 +1,176 @@
+import me.shedaniel.math.Rectangle;
+import me.shedaniel.rei.api.client.gui.Renderer;
+import me.shedaniel.rei.api.client.gui.widgets.Slot;
+import me.shedaniel.rei.api.client.gui.widgets.Widget;
+import me.shedaniel.rei.api.client.gui.widgets.Widgets;
+import me.shedaniel.rei.api.client.registry.display.DisplayCategory;
+import me.shedaniel.rei.api.common.category.CategoryIdentifier;
+import me.shedaniel.rei.api.common.entry.EntryIngredient;
+import me.shedaniel.rei.api.common.util.EntryStacks;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.client.gui.Element;
+import net.minecraft.client.render.DiffuseLighting;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.MathHelper;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author <a href="https://github.com/TeamGalacticraft">TeamGalacticraft</a>
+ */
+@Environment(EnvType.CLIENT)
+public class DefaultCompressingCategory implements DisplayCategory<DefaultCompressingDisplay> {
+
+    @Override
+    public CategoryIdentifier<? extends DefaultCompressingDisplay> getCategoryIdentifier() {
+        return GalacticraftREIClientPlugin.COMPRESSING;
+    }
+
+    @Override
+    public Renderer getIcon() {
+        return EntryStacks.of(new ItemStack(GalacticraftBlock.COMPRESSOR));
+    }
+
+    @Override
+    public Text getTitle() {
+        return new TranslatableText("category.rei.compressing");
+    }
+
+    public @NotNull List<Widget> setupDisplay(DefaultCompressingDisplay recipeDisplay, Rectangle bounds) {
+        final Point startPoint = new Point(bounds.getCenterX() - 68, bounds.getCenterY() - 37);
+
+        class BaseWidget extends Widget {
+            BaseWidget() {
+            }
+
+            public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
+                DiffuseLighting.disableGuiDepthLighting();
+                RenderSystem.setShaderTexture(0, Constant.ScreenTexture.REI_DISPLAY_TEXTURE);
+                this.drawTexture(matrices, startPoint.x, startPoint.y, 0, 83, 137, 157);
+
+                int height = MathHelper.ceil((double) (System.currentTimeMillis() / 250L) % 14.0D);
+                this.drawTexture(matrices, startPoint.x + 2, startPoint.y + 21 + (14 - height), 82, 77 + (14 - height), 14, height);
+                int width = MathHelper.ceil((double) (System.currentTimeMillis() / 250L) % 24.0D);
+                this.drawTexture(matrices, startPoint.x + 24, startPoint.y + 18, 82, 91, width, 17);
+            }
+
+            @Override
+            public List<? extends Element> children() {
+                return Collections.emptyList();
+            }
+        }
+
```

Comment:
Do we want to export this to its own class or as an inner class?

Reply:
Tbh we might not even need it

Reply:
done in 84de190